quickwit-cli/tests/cli.rs

```diff
@@ -0,0 +1,243 @@
+    .stdout(predicate::str::contains("Indexed"))
+    .stdout(predicate::str::contains("documents in"))
+    .stdout(predicate::str::contains(
+        "You can now query your index with",
+    ));
+}
+
+fn delete_index(test_env: &TestEnv) {
+    make_command(format!("delete --index-uri {} ", test_env.uri).as_str())
+        .assert()
+        .success();
+    assert_eq!(test_env.path.exists(), false);
+}
+
+#[test]
+fn test_cmd_help() -> anyhow::Result<()> {
+    let mut cmd = make_command("--help");
+    cmd.assert()
+        .success()
+        .stdout(predicate::str::contains("USAGE"));
+    Ok(())
+}
+
+#[test]
+fn test_cmd_new() -> Result<()> {
+    let test_env = create_test_env(false)?;
+    create_wikipedia_index(&test_env);
+
+    // Attempt to create with ill-formed new command.
+    make_command("new")
+        .assert()
+        .failure()
+        .stderr(predicate::str::contains("--index-uri <INDEX URI>"));
+
+    Ok(())
+}
+
+#[test]
+fn test_cmd_new_on_existing_index() -> Result<()> {
+    let test_env = create_test_env(false)?;
+    create_wikipedia_index(&test_env);
+
+    make_command(
+        format!(
+            "new --index-uri {} --doc-mapper-type wikipedia",
+            test_env.uri
+        )
+        .as_str(),
+    )
+    .assert()
+    .failure()
+    .stderr(predicate::str::contains("already exists"));
+
+    Ok(())
+}
+
+#[test]
+fn test_cmd_index() -> Result<()> {
+    let test_env = create_test_env(false)?;
+    create_logs_index(&test_env);
+
+    // Indexing data on non-existing index.
+    make_command(
+        format!(
+            "index --index-uri {}/non-existing-index --input-path {}",
+            test_env.uri,
+            test_env.log_data_path.display(),
+        )
+        .as_str(),
+    )
+    .assert()
+    .failure()
+    .stderr(predicate::str::contains("does not exist"));
+
+    // Index on non-existing data file.
+    make_command(
+        format!(
+            "index --index-uri {} --input-path {}",
+            test_env.uri,
+            test_env.dir.path().join("non-existing-data.json").display(),
+        )
+        .as_str(),
+    )
+    .assert()
+    .failure()
+    .stderr(predicate::str::contains("No such file or directory"));
+
+    index_data(&test_env.uri, test_env.log_data_path.as_path());
+
+    // Indexing data with piped input.
+    make_command(format!("index --index-uri {} ", test_env.uri).as_str())
+        .pipe_stdin(test_env.log_data_path)?
+        .assert()
+        .success()
+        .stdout(predicate::str::contains("Indexed"))
+        .stdout(predicate::str::contains("documents in"))
+        .stdout(predicate::str::contains(
+            "You can now query your index with",
+        ));
+
+    Ok(())
+}
+
+#[test]
+fn test_cmd_delete() -> Result<()> {
+    let test_env = create_test_env(false)?;
+    create_wikipedia_index(&test_env);
+
+    // Delete empty index.
+    make_command(format!("delete --index-uri {} --dry-run", test_env.uri).as_str())
```

Comment:
why using dry-run ?

Reply:
I am asserting on the output of the execution to see if the correct message is shown to the user. since the actual deletion is tested later at `delete_index(&test_env);`

Reply:
indeed, so the name `delete_index` is confusing as it's also doing some assertions. 

Can you rename it and do the same for all functions that do some assertions (renaming like delete_index_and_check_success would be sufficient) ? Another possibility is to do the assertions in the test directly.

Reply:
renaming is good but `delete_index_and_check_success` is not helping.

`delete_index_dry_run` would be nice.